connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -178,6 +178,7 @@ private List<SourceRecord> sourceRecordsForGroup(String group) throws Interrupte
         }
     }
 
+    // There may be a group that consumes the topic in "topics.exclude", so it also needs to be filtered
```

Comment:
Isn't this fairly redundant? We already filter the results for `shouldCheckpointTopic`, and I don't really see how the PR here makes that any more/less likely.

Also, if we really need a comment here, it shouldn't refer to `topics.exclude` (since that property is specific to the `DefaultTopicFilter`) and should instead just refer to whether or not the user-configured topic filter accepts the topic(s).

Reply:
> 

1. In MirrorCheckpointConnector, the group list is obtained by filtering through "shouldReplicateByTopicFilter". Therefore, compared with the trunk, every MirrorCheckpointTask is allocated to fewer groups (that is, it solves KAFKA-14842)
2. Because the principle of filtering groups through "shouldReplicateByTopicFilter" in MirrorCheckpointConnector is: as long as there is at least one topic accepted by topicFilter, the group will be assigned to MirrorCheckpointTask for replication, but the group may also consume other topics that are not accepted by topicFilter at the same time, so The offsets corresponding to these unaccepted topics should not be replicated.(because the syncGroupOffset thread takes the data from "checkpointsPerConsumerGroup", so it needs to be filtered by topic filter again.)
@C0urante

Reply:
I re-edited the comments here.